rllib/algorithms/algorithm.py

```diff
@@ -2179,6 +2186,28 @@ def default_resource_request(
 
         bundles += rollout_workers + evaluation_bundle
 
+        if cf._enable_rl_trainer_api:
+            # resources for the trainer
+            if cf.num_trainer_workers == 0:
+                # if num_trainer_workers is 0, then we need to allocate one gpu if
+                # num_gpus_per_trainer_worker is greater than 0.
+                trainer_bundle = [
+                    {
+                        "CPU": cf.num_cpus_per_trainer_worker,
+                        "GPU": int(cf.num_gpus_per_trainer_worker > 0),
```

Comment:
ok. I just want to provide my personal perspective here.
I feel like we always try to "auto-correct" for our users. for example, compute and override some parameters based on values of other parameters. we also put up a lot of explanations / docs around the "corrections" we may potentially do.
if it's 100% up to me, I would instead just raise error and telling users that I am seeing contradictory configs, because ``num_trainer_workers=0`` doesn't work with ``num_gpus_per_trainer_worker > 0``. now, we have no idea what the actual user intention is. did they mis-specify ``num_trainer_workers`` or did they mis-specify ``num_gpus_per_trainer_worker``? the only one who can fix this is the user him/herself.
now, this is just some of my thoughts. I can't enforce this, so hopefully you understand what I meant.

Reply:
got it. make sense.

Reply:
done